rs/nns/integration_tests/src/node_provider_remuneration.rs

```diff
@@ -216,7 +217,63 @@ fn test_list_node_provider_rewards() {
     // check rewards are as expected
     assert_eq!(
         response.rewards,
-        minted_rewards[8..13]
+        minted_rewards[..].iter().rev().cloned().collect::<Vec<_>>()
+    );
+
+    // Next we test the date filter with no start_date
+    let response = nns_list_node_provider_rewards(
+        &state_machine,
+        ListNodeProviderRewardsRequest {
+            date_filter: Some(DateRangeFilter {
+                start_timestamp_seconds: None,
+                end_timestamp_seconds: Some(minted_rewards_timestamps[11]),
+            }),
+        },
+    );
+    let received_ts: Vec<u64> = response.rewards.iter().map(|r| r.timestamp).collect();
+    assert_eq!(
+        received_ts,
+        minted_rewards_timestamps[0..12]
```

Comment:
NCR. I feel it's slightly better to do `0..=11` since 11 appears above

Reply:
done